pkg/cloudprovider/aws.go

```diff
@@ -0,0 +1,107 @@
+package cloudprovider
+
+import (
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/opsworks"
+	"github.com/sirupsen/logrus"
+
+	"github.com/libopenstorage/cloudops"
+	awsops "github.com/libopenstorage/cloudops/aws"
+)
+
+type aws struct {
+	defaultProvider
+
+	ops cloudops.Ops
+}
+
+func (a *aws) init() error {
+	// Do parent init first
+	if err := a.defaultProvider.init(); err != nil {
+		return err
+	}
+	if a.distributionName != eksDistribution {
+		return nil
+	}
+
+	// Initialize aws client for cloud drive permission checks
+	client, err := awsops.NewClient()
+	if err != nil {
+		return err
+	}
+	a.ops = client
+	return nil
+}
+
+func (a *aws) Name() string {
+	return string(cloudops.AWS)
+}
+
+func (a *aws) CheckCloudDrivePermission() error {
+	// check the permission here by doing dummy drive operations
+	logrus.Infof("JLIAO: doing aws permisison check")
+	volTypeGp2 := opsworks.VolumeTypeGp2
+	volID := "px-check-aws-cloud-permission"
+	volSize := int64(1)
+
+	// Create
+	v, err := a.ops.Create(&ec2.Volume{
```

Comment:
We should handle already exists error gracefully and not fail. It could be a common issue during retries where the volume created succeeded but other operations failed.